okhttp/src/main/java/okhttp3/internal/http2/Http2Connection.java

```diff
@@ -755,13 +741,13 @@ private void applyAndAckSettings(final Settings peerSettings) {
 
     @Override public void ping(boolean reply, int payload1, int payload2) {
       if (reply) {
-        Ping ping = removePing(payload1);
-        if (ping != null) {
-          ping.receive();
+        synchronized (Http2Connection.this) {
+          receivedPongCount++;
```

Comment:
Is there value in checking payload1 and payload2 in this case? e.g. whether it is a reply to a PING, or a standalone keepalive PING ACK?  Seems like that is a possibility, but I'm not very familiar with these in practice.